app/models/piece.rb

```diff
@@ -2,75 +2,62 @@
-            @pieces_in_the_way << chess_piece
-          end
-        end
-      end
-      !@pieces_in_the_way.empty?
+      chess_piece.y_position.between?(y_sorted_array[0] + 1, y_sorted_array[1] - 1)
 
-    #Horizontal
-    elsif x == self.x_position 
-      @pieces_in_the_way = []
-      self.game.pieces.each do |chess_piece|    if chess_piece != self
-          if (chess_piece.x_position == x) && chess_piece.y_position.between?(y_sorted_array[0], y_sorted_array[1])
-            @pieces_in_the_way << chess_piece
-          end
-        end
-      end
-      !@pieces_in_the_way.empty?
-
-    #Diagonal
-    elsif (x - self.x_position).abs == (y-self.y_position).abs
-      @pieces_in_the_way = self.game.pieces.select do |chess_piece|
-        next false if chess_piece == self
-        ((x - chess_piece.x_position).abs == (y - chess_piece.y_position).abs) && chess_piece.x_position.between?(x_sorted_array[0], x_sorted_array[1]) && chess_piece.y_position.between?(y_sorted_array[0], y_sorted_array[1])
-      end
-      !@pieces_in_the_way.empty?
+    end
 
+    return obstructions.present? || players_own_piece_is_there?(x, y) 
+    
+  end
+  
+  def has_diagonal_obstruction?(x, y, chess_piece, x_sorted_array, y_sorted_array)
+    is_eq_abs = (x - chess_piece.x_position).abs == (y - chess_piece.y_position).abs
+    is_between_x = chess_piece.x_position.between?(x_sorted_array[0] + 1, x_sorted_array[1] - 1)
+    is_between_y = chess_piece.y_position.between?(y_sorted_array[0] + 1, y_sorted_array[1] - 1)
+    return is_eq_abs && is_between_x && is_between_y
+  end
+  
+  def players_own_piece_is_there?(x, y)
+    occupying_piece = Piece.where(x_position: x, y_position: y, game_id: game.id)
+    if occupying_piece.any? then
+      return occupying_piece.first.is_white? && is_white? || occupying_piece.first.is_black? && is_black?
     else
       false
     end
   end
 
-  def move_to!(new_x, new_y)
-    # is valid
-    # is obstructed
-
-    # Check if a piece is at this position
-    if Piece.where(x_position: new_x, y_position: new_y, game_id: self.game.id)[0]
-      occupying_piece = Piece.where(x_position: new_x, y_position: new_y, game_id: self.game.id)[0]
-
-      # Check if the piece occupying the space is white(0-5) and the moving piece is black(6-11)
-      if occupying_piece.piece_number > 5 && self.piece_number < 6
-        occupying_piece.update(x_position: 8, y_position: 0)
-        self.update(x_position: new_x, y_position: new_y)
-
-      # Check if the piece occupying the space is black(6-11) and the moving piece is white(0-5)
-      elsif occupying_piece.piece_number < 6 && self.piece_number > 5
-        occupying_piece.update(x_position: 9, y_position: 0)
-        self.update(x_position: new_x, y_position: new_y)
-      else
-        ActionDispatch::Flash.new(alert: 'You cannot move here.')
-      end
+  def move_to!(x,y) 
```

Comment:
Love the simplification of this method 100%
It really seems to do what you would think from the name now.

Reply:
Yes my mentor was fairly upset that we had a method doing all that it was doing.